### PR TITLE
Fix: p2p trade history counter party value

### PR DESCRIPTION
**Pull request by dhanuwat**
[Jira Ticket](https://openware-inc.atlassian.net/browse/NAIR-210)

Fix incorrect user displayed in counterparty column in P2P history table.

### CODE REVIEW

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ const TradesHistory: FC = (): ReactElement => {
     };
 
     const renderContent = () => {
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:62]: use React.useMemo

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ const TradesHistory: FC = (): ReactElement => {
                     />}
                 </div>
             </div>
-        ];
+        );
     };
 
     const retrieveData = () => {
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:89]: use React.useMemo

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 1 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 2 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

**Review by josadcha — APPROVED** [review]

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 3]
```diff
@@ -40,6 +40,8 @@ const TradesHistory: FC = (): ReactElement => {
     const nextPageExists = useSelector((state: RootState) => selectP2PTradesHistoryNextPageExists(state, DEFAULT_TABLE_PAGE_LIMIT));
     const currencies = useSelector(selectCurrencies);
 
+    const translate = useCallback((id: string) => formatMessage({ id: id }), [formatMessage]);
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:43]: 1. { id }
2. Remove format message

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 4]
```diff
@@ -48,29 +50,56 @@ const TradesHistory: FC = (): ReactElement => {
...
-            intl.formatMessage({id: 'page.body.p2p.order_history.status'}),
+            translate('page.body.p2p.order_history.date'),
+            translate('page.body.p2p.order_history.side'),
+            translate('page.body.p2p.order_history.price'),
+            translate('page.body.p2p.order_history.quantity'),
+            translate('page.body.p2p.order_history.counterparty'),
+            translate('page.body.p2p.order_history.status'),
         ];
-    };
+    }, [translate]);
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:62]: Remove translate here

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 5]
```diff
@@ -48,29 +50,56 @@ const TradesHistory: FC = (): ReactElement => {
...
-    const renderContent = () => {
         return [
+            localeDate(created_at, 'fullDate'),
+            <span style={{ color: sideColored.color }}>{sideColored.text}</span>,
+            `${Decimal.format(price, pricePrecision || DEFAULT_FIAT_PRECISION, ',')} ${base.toUpperCase()}/${quote.toUpperCase()}`,
+            `${Decimal.format(amount, amountPrecision || DEFAULT_CCY_PRECISION, ',')} ${base.toUpperCase()}`,
+            <span>{user?.user_nickname || uid}</span>,
+            <span style={{ color: stateColored.color }}>{stateColored.text}</span>,
+        ];
+    }, [currencies, DEFAULT_FIAT_PRECISION, DEFAULT_CCY_PRECISION, setTradesType, setStateType]);
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:80]: Keep here only currencies

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 6]
```diff
@@ -48,29 +50,56 @@ const TradesHistory: FC = (): ReactElement => {
...
+
+    const retrieveData = useMemo(() => {
+        return [...list].map(item => renderTableRow(item));
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:83]: Why do you use ...list?
Do list.map(...)

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 7]
```diff
@@ -48,29 +50,56 @@ const TradesHistory: FC = (): ReactElement => {
...
+    }, [list, renderTableRow]);
+
+    const onRowClick = useCallback((index: string) => {
+        const orderId = list[index]?.id;
+        orderId && history.push(`/p2p/order/${orderId}`);
+    }, [history, list]);
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:89]: History is extra property 
Remove it

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 8]
```diff
@@ -83,35 +112,8 @@ const TradesHistory: FC = (): ReactElement => {
...
-            localeDate(created_at, 'fullDate'),
-            <span style={{ color: sideColored.color }}>{sideColored.text}</span>,
-            `${Decimal.format(price, pricePrecision || DEFAULT_FIAT_PRECISION, ',')} ${base.toUpperCase()}/${quote.toUpperCase()}`,
-            `${Decimal.format(amount, amountPrecision || DEFAULT_CCY_PRECISION, ',')} ${base.toUpperCase()}`,
-            <span>{user?.user_nickname || user_uid}</span>,
-            <span style={{ color: stateColored.color }}>{stateColored.text}</span>,
-        ];
-    };
+        );
+    }, [list, firstElemIndex, lastElemIndex, total, page, nextPageExists, onClickPrevPage, onClickNextPage, renderHeaders, retrieveData, onRowClick, translate]);
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:116]: Please check all these values here
I'm sure some of them shouldn't be here

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 9]
```diff
@@ -120,8 +122,8 @@ const TradesHistory: FC = (): ReactElement => {
     return (
         <div>
             {fetching && <div className="text-center"><Spinner animation="border" variant="primary" /></div>}
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:122]: Create a separate method using useMemo for this block

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 10]
```diff
@@ -120,8 +122,8 @@ const TradesHistory: FC = (): ReactElement => {
...
-            {list.length ? renderContent() : null}
-            {!list.length && !fetching ? <p className="pg-history-elem__empty">{intl.formatMessage({id: 'page.noDataToShow'})}</p> : null}
+            {list.length ? renderContent : null}
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:125]: Create a separate method using useMemo for this block

File: web/src/containers/P2P/TradesHistory/index.tsx [hunk 11]
```diff
@@ -120,8 +122,8 @@ const TradesHistory: FC = (): ReactElement => {
...
+            {!list.length && !fetching ? <p className="pg-history-elem__empty">{translate('page.noDataToShow')}</p> : null}
```
Comment (akhlopiachyi) [inline web/src/containers/P2P/TradesHistory/index.tsx:126]: Create a separate method using useMemo for this block

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 3 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 5 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 7 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 9 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx:122]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 11 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 10 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 8 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 6 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

File: web/src/containers/P2P/TradesHistory/index.tsx (hunk 4 above)
Comment (dhanuwat) [reply web/src/containers/P2P/TradesHistory/index.tsx]: Fixed, thanks.

**Review by akhlopiachyi — APPROVED** [review]

**Pull request merged**